### ISSUE TITLE
Exception in thread Thread-2 when sending yowsup 3.0.0

### ISSUE THREAD

**Issue by epizarro2** [tgalal/yowsup#2670]
For reproduce:

1. fresh install debian 9.8
2. apt install python python-pip openssl
3. pip install yowsup
4. read the short yowsup-cli manual

```
root@debian:~# yowsup-cli registration --register WWW-WWW --config-phone 56XXXXXXXXX
yowsup-cli  v3.0.0
yowsup      v3.0.0

Copyright (c) 2012-2019 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


I 2019-04-23 23:05:16,672 yowsup.common.http.warequest - {"status":"ok","login":"56XXXXXXXXX","type":"existing","edge_routing_info":"CAUIAg==","chat_dns_domain":"fb","security_code_set":false}

{
    "__version__": 1,
    "cc": "56",
    "client_static_keypair": "ZzZzZZZzzZzZzzZZzZzZZZZzZzZZZzzZzZzz+Z/ZzZZZZzZzZZZzzZzZzzZZzZzZZZZzZzZZZzzZzZzzZZzZzZ==",
    "expid": "ZzZzZZZzzZzZzzZZzZzZZZ==",
    "fdid": "ZZZzzZzZ-ZZZz-ZZZz-ZZZz-ZZZzzZzZzZzZ",
    "id": "ZZZzzZzZzZzZ+ZZZzzZzZzZzZzZ=",
    "mcc": "730",
    "mnc": "08",
    "phone": "56XXXXXXXXX ",
    "sim_mcc": "000",
    "sim_mnc": "000"
}
status: ok
edge_routing_info: CAUIAg==
login: 56XXXXXXXXX
type: existing
```

```
root@debian:~# ls -lah .config/yowsup/56XXXXXXXXX/
total 148K
drwxr-xr-x 2 root root 4.0K Apr 23 23:18 .
drwxr-xr-x 3 root root 4.0K Apr 23 23:12 ..
-rw-r--r-- 1 root root 132K Apr 23 23:18 axolotl.db
-rw-r--r-- 1 root root  440 Apr 23 23:13 config.json
```

```
root@debian:~# yowsup-cli demos --config-phone 56XXXXXXXXX --send 56YYYYYYYYY 'hi'
yowsup-cli  v3.0.0
yowsup      v3.0.0

Copyright (c) 2012-2019 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


I 2019-04-23 23:18:12,357 yowsup.layers.network.layer - Connecting to e4.whatsapp.net:443
I 2019-04-23 23:18:12,570 yowsup.axolotl.manager - Generating 812 prekeys
I 2019-04-23 23:18:12,733 yowsup.axolotl.manager - Storing 812 prekeys
I 2019-04-23 23:18:21,880 yowsup.axolotl.manager - Loaded 812 unsent prekeys
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/dist-packages/yowsup/layers/noise/workers/handshake.py", line 36, in run
    self._protocol.start(self._stream, self._client_config, self._s, self._rs)
  File "/usr/local/lib/python2.7/dist-packages/consonance/protocol.py", line 82, in start
    result = handshake.perform(client_config, stream, s, rs)
  File "/usr/local/lib/python2.7/dist-packages/consonance/handshake.py", line 84, in perform
    cipherstatepair = self._start_handshake_xx(stream, client_payload, dissononce_s)
  File "/usr/local/lib/python2.7/dist-packages/consonance/handshake.py", line 110, in _start_handshake_xx
    self._handshakestate.write_message(b'', ephemeral_public)
  File "/usr/local/lib/python2.7/dist-packages/dissononce/extras/processing/handshakestate_forwarder.py", line 16, in write_message
    return self._handshakestate.write_message(payload, message_buffer)
  File "/usr/local/lib/python2.7/dist-packages/dissononce/extras/processing/handshakestate_guarded.py", line 90, in write_message
    self._handshake_machine.next()
  File "/usr/local/lib/python2.7/dist-packages/future/types/newobject.py", line 54, in next
    raise TypeError('newobject is not an iterator')
TypeError: newobject is not an iterator

^C
Yowsdown
```

**Comment by tgalal** [comment]
This is due to a bug in dissononce, (https://github.com/tgalal/dissononce/issues/1) I'll release updates soon fixing the issue.

**Comment by tgalal** [comment]
fixed in [v3.1.0](https://github.com/tgalal/yowsup/releases/tag/v3.1.0)

**Comment by epizarro2** [comment]
thanks that was quick. take my money 💃
